### PR TITLE
Feature/facetconfig

### PR DESCRIPTION
**Pull request by jariz**
How to test:
http://localhost:3000/edit/u33707283d426f900d4d33707283d426f900d4d0d__clusius2/facet-config/clusius_Places

What works:
- Loading data from graph into reducer
- Adding
- Sorting

What still to do:
- Choosing different facet type
- Configuring facet paths
- Submitting facet mutation

### CODE REVIEW

**Comment by jauco** [comment]
It seems to add an empty item to the top of the list (so if you have 4 items configured you get 5, first one empty)

File: src/reducers/facetconfig.ts [hunk 1]
```diff
@@ -0,0 +1,149 @@
+import { FacetConfig } from '../typings/schema';
+import { NormalizedFacetConfig } from '../typings/index';
+import { arrayMove } from 'react-sortable-hoc';
+import { EMPTY_FACET_CONFIG } from '../constants/global';
+
+// state def
+export type FacetConfigReducer = NormalizedFacetConfig[];
+const defaultState: FacetConfigReducer = [{ id: 0, ...EMPTY_FACET_CONFIG }];
```
Comment (jariz) [inline src/reducers/facetconfig.ts]: @jauco Dat komt door deze regel denk ik, moet een lege array zijn.

**Comment by jariz** [comment]
@jauco Feedback processed

File: src/components/form/DraggableForm.tsx [hunk 2]
```diff
@@ -117,15 +124,46 @@ class DraggableForm extends PureComponent<Props, State> {
...
+            sortItem: (oldIndex: number, newIndex: number) => dispatch(sortFacetConfigItem(oldIndex, newIndex)),
+        };
+    }
+};
+
+const mergeProps = (stateProps: StateProps, dispatchProps: DispatchProps, ownProps: OwnProps): Props => ({
+    ...stateProps,
+    ...dispatchProps,
+    ...ownProps,
+    addItem: (item: ConfigurableItem) => {
```
Comment (ichsaras) [inline src/components/form/DraggableForm.tsx:146]: Aren't you overwriting the addItem declared in mapDispatchToProps for the same check (configType === 'view')?

File: src/reducers/facetconfig.ts [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+    switch (action.type) {
+        case 'ADD_FACET_CONFIG_ITEM':
+            return {
+                id: lastId(items) + 1,
+                ...action.payload.facetConfig
+            };
+        case 'MODIFY_FACET_CONFIG_ITEM':
+            return {
+                id: state!.id,
+                ...action.payload.facetConfig
```
Comment (ichsaras) [inline src/reducers/facetconfig.ts]: Spread operators should always be above the new value you're providing, just to make sure it gets overwritten by the new value your pushing right?

**Review by ichsaras — APPROVED** [review]

File: src/components/form/DraggableForm.tsx (hunk 2 above)
Comment (jariz) [reply src/components/form/DraggableForm.tsx:146]: Yea, this is by design.
dispatchProps's addItem only adds an item, this overload is there to also add it is a child immediately so the component does not need to do that itself.
I wanted to put as much of the logic as possible into the mapping functions so the component itself can be agnostic to what type of item it's actually editing.

**Pull request merged**